packages/core/core.test.mts

```diff
@@ -83,6 +83,31 @@ describe('createPrompt()', () => {
     await expect(answer).resolves.toEqual('up');
   });
 
+  it('useEffect: works with setting state at once with objects', async () => {
+    const Prompt = (config: { message: string }, done: (value: string) => void) => {
+      const [value, setValue] = useState([1, 2]);
+
+      useEffect(() => {
+        setValue([1, 3]);
+      }, []);
+
+      useKeypress((key: KeypressEvent) => {
+        if (isEnterKey(key)) {
+          done(String(value));
+        }
+      });
+
+      return String(value);
+    };
+
+    const prompt = createPrompt(Prompt);
+    const { answer, events } = await render(prompt, { message: 'Question' });
+    events.keypress('enter');
+
+    const resolvedAnswer = await answer;
```

Comment:
awaiting it instead of using expect(answer).resolves as this produces a better error message.

Reply:
Mind adding an inline comment so we don't loose that context 😄

Reply:
done :)